File: examples/python/drop.py
"""
Conditionally drop requests and responses
"""

REQ_FILTER = """
(path-matches r"^/logout$")
"""

RES_FILTER = """
(path-matches r"^/login$")
"""

def on_request(req):
    log.info("Dropping logout")
    return req.drop()

def on_response(res):
    try:
        obj = res.bodyToJson()
        if obj.get("status") == "failed":
            log.info("Dropping failed login response")
            return res.drop()
    except Exception as e:
        pass

    return res

File: examples/js/drop.mjs
/**
 * Conditionally drop requests and responses
 */

export const REQ_FILTER = `(path-matches r"^/logout$")`
export const RES_FILTER = `(path-matches r"^/login$")`

export function onRequest(req) {
    log.info("Dropping logout")
    return req.drop()
}

export function onResponse(res) {
    try {
        const obj = res.bodyToJson();
        if (obj.status == "failed") {
            log.info("Dropping failed login response")
            return res.drop()
        }
    } catch (e) {
        log.error(e)
    }

    return res
}
